it/tutorials/scala-for-java-programmers.md

```diff
@@ -0,0 +1,716 @@
+---
+layout: overview
+title: Un'introduzione a Scala per programmatori Java
+overview: scala-for-java-programmers
+
+disqus: true
+multilingual-overview: true
+language: [it]
+---
+
+Di Michel Schinz e Philipp Haller.
+Traduzione italiana a cura di Mirco Veltri.
+
+## Introduzione
+
+Lo scopo di questo documento è quello di fornire una rapida
+introduzione al linguaggio e al compilatore Scala. È rivolto a
+chi ha già qualche esperienza  di programmazione e desidera una
+panoramica di cosa è possibile fare con Scala. Si assume una
+conoscenza di base dei concetti di programmazione orientata
+agli oggetti, specialmente in Java.
+
+## Un Primo Esempio
+
+Come primo esempio useremo lo standard *Hello world*.  Non è sicuramente
+un esempio entusiasmante ma rende facile dimostrare l'uso dei tool di
+Scala senza richiedere troppe conoscenze del linguaggio stesso.
+Ecco come appeare il codice:
+
+    object HelloWorld {
+      def main(args: Array[String]) {
+        println("Hello, world!")
+      }
+    }
+
+La struttura di questo programma dovrebbe essere familiare ai
+programmatori Java: c'è un metodo chiamato `main` che accetta argomenti,
+un array di stringhe, forniti da riga di comando come
+parametri; Il corpo del metodo consiste di una singola chiamata al
+predefinito `println` che riceve il nostro amichevole saluto
+come parametro. Il metodo `main` non ritorna alcun valore
+(è un metodo procedura), pertanto non è necessario dichiararne uno
+di ritorno.
+
+Ciò che è meno familiare ai programmatori Java è la
+dichiarazione di `object` contenente il metodo `main`.
+Tale dichiarazione introduce ciò che è comunemente chiamato
+*oggetto singleton*, cioè una classe con una unica istanza.
+La dichiarazione precedente infatti crea sia la classe `HelloWorld`
+che una istanza di essa, chiamata `HelloWorld`. L'istanza è creata
+su richiesta la prima volta che è usata.
+
+
+Il lettore astuto avrà notato che il metodo `main` non è stato
+dichiarato come `static`. Questo perchè  i membri (metodi o campi)
+statici non esistono in Scala. Invece che definire membri statici,
+il programmatore Scala li dichiara in oggetti singleton.
+
+### Compiliamo l'esempio
+
+Per compilare l'esempio useremo `scalac`, il compilatore Scala.
+`scalac` lavora come la maggior parte dei compilatori: prende un file
+sorgente come argomento, eventuali opzioni e produce  uno o più object
+file come output. Gli object file sono gli standard file delle classi
```

Comment:
s/object file/class file

Also considering that in the next paragraph `class file` is used already.